### DISCUSSION TITLE
ValueError: Can't read file: patterns\amt_patterns2021-10-12.jsonl\cfg

### DISCUSSION THREAD

**Discussion by robertpfaff** [explosion/spaCy#9485] [Help: Coding & Implementations] [answered]
I would appreciate some assistance with understanding this error message. I am using three entity rulers in this code. Their patterns files seem identical and were produced in the same way.  Their details and settings are the same. But this is the only patterns file (json lines file) giving me an error message. 

Strangely, I cannot find any reference to this error message elsewhere.

Thank you in advance for your sincere feedback. 

```
file_path = force_path(path)
File "C:\Users\rober\AppData\Local\Programs\Python\Python39\lib\site-packages\srsly\util.py", line 24, in force_path
raise ValueError(f"Can't read file: {location}")
ValueError: Can't read file: patterns\amt_patterns2021-10-12.jsonl\cfg
```

**Comment by kinghuang** [comment, marked as answer]
Do you have an extra path delimiter at the end of the path you're passing to [EntityRuler.from_disk](https://spacy.io/api/entityruler#from_disk)? It seems to be treating whatever you're passing in as a directory, and is trying to read the entity ruler's config file within as a result.

```python
# NO
entity_ruler.from_disk("patterns/amt_patterns2021-10-12.jsonl/")

# YES
entity_ruler.from_disk("patterns/amt_patterns2021-10-12.jsonl")
```

**Reply by robertpfaff** [reply]
Thank you for your response.The whole line block of code looks like this:

```
nlp = spacy.load("en_core_web_lg")

rulerIngs = nlp.add_pipe("entity_ruler", name="rulerIngs", before="ner")
rulerUms = nlp.add_pipe("entity_ruler", name="rulerUms", before="ner")
rulerAmt = nlp.add_pipe("entity_ruler", name="rulerAmt", before="ner")

rulerIngs.from_disk("patterns/ing_patterns2021-10-11.jsonl")
rulerUms.from_disk("patterns/ums_patterns2021-10-11.jsonl")
rulerAmt.from_disk("patterns/amt_patterns2021-10-12.jsonl")
```

So there is no difference between it and the other entity rulers, except that it's the last one. But its place in line is not the problem. If I comment it out, the other rulers still work fine. I tried playing with the file name, and the file location, but it doesn't seem to make a difference. I am going to generate it using Prodigy this morning. Before I had Prodigy, I wrote a piece of code that created the patterns files. And it works just fine. I don't know why the system expects to find a config file. I have a config and base_config file in the main folders, set to defaults. And there is no apparent, direct relationship between those files and these files. Or none that I could find. So we'll see. Thanks again.

P.S. It is working now, after I created and exported the patterns file through Prodigy using the NER manual and terms.to-patterns module. But that should not be necessary and I am still curious as possible causes of the problem.

**Reply by kinghuang** [reply]
Weird. At least you got it to work!

**Comment by polm** [comment]
Does the file actually exist? I know it's a silly thing, but just to be sure... One thing is that the filename you give ends in a 12 for that component but 11 for the others, is that correct?

Is it possible you have a failing SSD, or some network share status that makes the file behave strangely?

It looks like you're using forward slashes in code, but backslashes are showing up in your traceback. This is a longshot, but maybe `\a` is being interpreted as an ASCII bell somewhere - if you change the filename to something like "iamt" does it work?

Those are all kind of longshots, but file writing and reading is rather low-level, so if the problem is there it's hard to be sure what's happening without being able to check your system.
